salt/modules/rpm.py

```diff
@@ -688,14 +688,29 @@ def _prepend(ver):
                     'more accurate version comparisons'
                 )
         else:
-            cmp_result = cmp_func(salt.utils.str_version_to_evr(ver1),
-                                  salt.utils.str_version_to_evr(ver2))
-            if cmp_result not in (-1, 0, 1):
-                raise CommandExecutionError(
-                    'Comparison result \'{0}\' is invalid'.format(cmp_result)
-                )
-
-            return cmp_result
+            else:
```

Comment:
This "else" directly follows the previous "else". There's no way this wouldn't raise an exception, was this tested?